File: server/common/factory.js
import userRepo from '../users/user.repo.js';
import UserService from '../users/user.service.js';
import projectRepo from '../project/project.repo.js';
import ProjectService from '../project/project.service.js';
import taskRepo from '../task/task.repo.js';
import TaskService from '../task/task.service.js';

export function getUserService() {
    return new UserService(userRepo);
}

export function getProjectService() {
    return new ProjectService(projectRepo);
}

export function getTaskService() {
    return new TaskService(taskRepo);
}

File: pkg/app/factory.go
package app

import (
	p "github.com/keuller/multi-task-manager/pkg/domain/project"
	t "github.com/keuller/multi-task-manager/pkg/domain/task"
	u "github.com/keuller/multi-task-manager/pkg/domain/user"
)

func BuildUserService() u.UserService {
	repo := u.NewUserRepository()
	return u.NewUserService(repo)
}

func BuildProjectService() p.ProjectService {
	repo := p.NewProjectRepository()
	return p.NewProjectService(repo)
}

func BuildTaskService() t.TaskService {
	repo := t.NewTaskRepository()
	return t.NewTaskService(repo)
}
